pkg/agent/util/net_windows.go

```diff
@@ -950,7 +960,173 @@ func RenameInterface(from, to string) error {
+		klog.InfoS("New VM Switch created", "switch", LocalVMSwitch)
+	}
+
+	enabled, err := isOVSExtensionEnabled()
+	if err != nil {
+		return err
+	}
+	if !enabled {
+		klog.InfoS("Enabling Open vSwitch Extension", "switch", LocalVMSwitch)
+		err := enableOVSExtension()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func RemoveVMSwitch() (bool, error) {
+	klog.InfoS("Deleting VmSwitch", "switch", LocalVMSwitch)
+	exists, err := VMSwitchExists()
+	if exists {
+		cmd := fmt.Sprintf(`Remove-VMSwitch -Name "%s" -ComputerName $(hostname) -Force`, LocalVMSwitch)
+		_, err := ps.RunCommand(cmd)
+		if err == nil {
+			return true, nil
+		}
+		if strings.Contains(err.Error(), fmt.Sprintf(`unable to find a virtual switch with name "%s"`, LocalVMSwitch)) {
+			return false, nil
+		}
+	}
+	return false, err
+}
+
+// createVMSwitchWithTeaming, creates VMSwitch and enables ovs extension
+// connection to vm is lost for few seconds
+func createVMSwitchWithTeaming(switchName, ifName string) error {
+	cmd := fmt.Sprintf(`New-VMSwitch -Name "%s" -NetAdapterName "%s" -EnableEmbeddedTeaming $true -AllowManagementOS $true -ComputerName $(hostname)| Enable-VMSwitchExtension "Open vSwitch Extension"`, switchName, ifName)
+	_, err := ps.RunCommand(cmd)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func enableOVSExtension() error {
+	cmd := fmt.Sprintf(`Get-VMSwitch -Name "%s" -ComputerName $(hostname)| Enable-VMSwitchExtension "Open vSwitch Extension"`, LocalVMSwitch)
+	_, err := ps.RunCommand(cmd)
+	if err != nil {
+		return err
+	}
 	return nil
 }
+
+/*
+Sample output of VM Switch extensions
+Id                  : 583CC151-73EC-4A6A-8B47-578297AD7623
+Name                : Open vSwitch Extension
+Vendor              : The Linux Foundation (R)
+Version             : 2.14.1.41988
+ExtensionType       : Forwarding
+ParentExtensionId   :
+ParentExtensionName :
+SwitchId            : 1cb724fe-0c01-42f2-a0e3-a1995676644a
+SwitchName          : sw-int
+Enabled             : False
+Running             : True
+CimSession          : CimSession: .
+ComputerName        : WIN2019
+IsDeleted           : False
+*/
+func parseOVSExtensionOutput(s string) bool {
+	scanner := bufio.NewScanner(strings.NewReader(s))
+	var forwardingExtension = false
+	for scanner.Scan() {
+		temp := strings.Fields(scanner.Text())
+		line := strings.Join(temp, "")
+		if strings.Contains(line, "Name") {
+			if strings.Contains(line, "OpenvSwitchExtension") {
+				klog.V(2).InfoS("Open vSwitch Extension is installed")
+				forwardingExtension = true
+			}
+		}
+		if forwardingExtension && strings.Contains(line, "Enabled") {
+			if strings.Contains(line, "True") {
+				klog.InfoS("Open vSwitch Extension is enabled")
+				return true
+			}
+			klog.InfoS("Open vSwitch Extension is disabled")
+			return false
+		}
+	}
+	return false
+}
+
+func isOVSExtensionEnabled() (bool, error) {
+	cmd := fmt.Sprintf(`Get-VMSwitchExtension -VMSwitchName "%s" -ComputerName $(hostname)`, LocalVMSwitch)
```

Comment:
Use this command to focus on OVS Extension only?
```
cmd := fmt.Sprintf(`Get-VMSwitchExtension -VMSwitchName "%s" -ComputerName $(hostname) | ? Id -EQ "%s"`, LocalVMSwitch, OVSExtensionID)
```

Reply:
Let me try it out.